Update ibm_catalog.json

arya-girish-k opened the pull request:
### Description
Modified the `ibm_catalog.json` by adding variable declarations and updating  catalog type.
Issue : [#90](https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/issues/90)
<!--- Replace this text with a summary of the changes in this PR. Include why the changes are needed and context about the changes. List required dependencies. If there is a Git issue for the change, please link to it. --->

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
Modified the `ibm_catalog.json` by adding variable declarations and updating  catalog type.
<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

arya-girish-k commented:
/run pipeline

arya-girish-k commented:
- Added variable declaration

- Added custom configuration for region type variables,these variables are coming under required tab: 
eg:
<img width="850" alt="Screenshot 2025-02-05 at 1 15 50 PM" src="https://github.com/user-attachments/assets/c50ebcd3-5717-46b1-9eb1-e0c499b6d2d3" />
- Added Multiple choice for the variable `secrets_manager_endpoint_type`.
eg:
<img width="791" alt="Screenshot 2025-02-05 at 1 13 56 PM" src="https://github.com/user-attachments/assets/d9d247f0-381d-4f5e-956b-904426a2b338" />

maheshwarishikha commented on ibm_catalog.json:
```diff
@@ -60,7 +75,132 @@
+          },
+          {
+            "key": "watson_assistant_region",
+            "custom_config": {
+                "type": "vpc_region",
+                "grouping": "deployment",
+                "original_grouping": "deployment",
+                "config_constraints": {
+                    "generationType": "2"
+                }
```
I think, we discussed last that we cant use `vpc_region` type. Do you know when can we use vpc_region? Is it the right usage?

maheshwarishikha replied:
same for other places, wherever you have used vpc_region

ocofaigh replied:
right, each service will have its own list of supported regions. We can only use the VPC region picker for variables that are going to deploy a VPC. We have to use a custom dropdown and add only the regions supported by that service

arya-girish-k replied:
@ocofaigh @maheshwarishikha : Ok Understood.I will make the necessary changes.

arya-girish-k replied:
Added custom drop down according to each services.

ocofaigh commented on ibm_catalog.json:
```diff
@@ -44,6 +44,21 @@
+              "key": "prefix"
+            },
+            {
+              "key": "use_existing_resource_group"
```
In order to ensure this shows beside the `resource_group_name` input in the Projects UI, we need to mark it explicitly as required here. See https://github.ibm.com/GoldenEye/issues/issues/12542#issuecomment-103499900

arya-girish-k replied:
Added the same.

ocofaigh commented on ibm_catalog.json:
```diff
@@ -60,7 +75,250 @@
+              ]
+          },
+          {
+            "key": "toolchain_resource_group"
```
We should be able to use the resource group picker here since it is a required variable and will always need an existing resource group

arya-girish-k replied:
Added resource group picker

arya-girish-k commented:
@ocofaigh @maheshwarishikha : The region limitation are not available in the code and the description mentions for eg :-`The region where the toolchains previously created reside`
Here I have implemented custom drop downs for variables for region variables as per each services .
For example for the variable `toolchain_region`, added the supported regions of the service  `continous delivery`.

I would like to clarify that should I proceed with this approach, or should I remove the custom dropdown from the file?

ocofaigh commented:
I think we have no choice but to use a custom dropdown. Its means we need to maintain it, but we have no other choice if we want to consumer experience to be better (which it will be by seeing a dropdown)

arya-girish-k commented:
/run pipeline

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 2.6.22 :tada:

The release is available on:
- [GitHub release](https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/releases/tag/v2.6.22)
- `v2.6.22`

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
